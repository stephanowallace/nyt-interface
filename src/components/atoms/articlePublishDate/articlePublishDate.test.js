import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import ArticlePublishDate from '.';

describe('Article publish date', () => {
  const date = moment('2018-12-15T13:22:47-05:00').format('MMM DD, YYYY');

  it('renders default state', () => {
    const component = shallow(<ArticlePublishDate>{date}</ArticlePublishDate>);
    expect(component).toMatchSnapshot();
  });

  it('renders loader', () => {
    const component = shallow(<ArticlePublishDate loading>test</ArticlePublishDate>);
    expect(component).toMatchSnapshot();
  });

  it('renders child in an time tag', () => {
    const component = shallow(<ArticlePublishDate>{date}</ArticlePublishDate>);
    expect(component.find('time').text()).toBe(date);
  });
});
