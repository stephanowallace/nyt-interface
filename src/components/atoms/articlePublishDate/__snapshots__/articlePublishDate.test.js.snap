// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article publish date renders default state 1`] = `
<SecondaryText
  className=""
>
  <time
    className="publishDate"
  >
    Dec 15, 2018
  </time>
</SecondaryText>
`;

exports[`Article publish date renders loader 1`] = `
<span
  className=""
>
  <ContentLoader
    className=""
    height="11px"
    width="80px"
  />
</span>
`;
